SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosOfferScheduler.java

```diff
@@ -999,7 +1018,11 @@ private double calculateScore(
     return score;
   }
 
-  private SingularityMesosTaskHolder acceptTask(
+  // This method is synchronized to avoid resource reuse within a single offer.
+  // Decisions about resources to use, specifically ports are made during the buildTask call, however
+  // these resources aren't subtracted from the offer until the addMatchedTask call, making this method
+  // not thread safe
+  private synchronized void acceptTask(
```

Comment:
Would it be possible/preferable to synchronize on the offerHolder instead of the method, to avoid blocking acceptTask calls that are using different offers?